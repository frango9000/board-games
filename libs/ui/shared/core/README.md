# ui-shared-core

This library was generated with [Nx](https://nx.dev).

## Running unit tests

Run `nx test ui-shared-core` to execute the unit tests.
