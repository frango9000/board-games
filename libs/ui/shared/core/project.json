{
  "$schema": "../../../../node_modules/nx/schemas/project-schema.json",
  "projectType": "library",
  "sourceRoot": "libs/ui/shared/core/src",
  "prefix": "app",
  "targets": {
    "clean": {
      "executor": "@nrwl/workspace:run-commands",
      "options": {
        "commands": ["rimraf ./coverage/libs/ui/shared/core"]
      }
    },
    "test": {
      "executor": "@nrwl/jest:jest",
      "outputs": ["coverage/libs/ui/shared/core"],
      "options": {
        "jestConfig": "libs/ui/shared/core/jest.config.ts",
        "passWithNoTests": true
      }
    },
    "lint": {
      "executor": "@nrwl/linter:eslint",
      "options": {
        "lintFilePatterns": ["libs/ui/shared/core/src/**/*.ts", "libs/ui/shared/core/src/**/*.html"]
      }
    }
  },
  "tags": []
}
