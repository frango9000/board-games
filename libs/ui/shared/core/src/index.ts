export * from './lib/store/localization/localization.repository';
export * from './lib/store/localization/localization.store.model';
export * from './lib/store/localization/localization.action';

export * from './lib/store/theme/theme.repository';
export * from './lib/store/theme/theme.store.model';
export * from './lib/store/theme/theme.action';

export * from './lib/services/core.service';
export * from './lib/services/card-view/card-view-header.service.model';
export * from './lib/services/card-view/card-view-header.service';
export * from './lib/services/sidenav/sidenav.service';
export * from './lib/services/toolbar/breadcrumb.service';

export * from './lib/services/toolbar/search.service';
export * from './lib/services/toolbar/toolbar.service.model';
export * from './lib/services/toolbar/toolbar.service';

export * from './lib/modules/is-mobile';
export * from './lib/modules/ng-let';
export * from './lib/modules/context-menu';
export * from './lib/modules/core-context-menu';

export * from './index.testing';
