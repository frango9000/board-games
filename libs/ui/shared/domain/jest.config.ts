/* eslint-disable */
export default {
  displayName: 'ui-shared-domain',
  preset: '../../../../jest.preset.js',
  setupFilesAfterEnv: ['<rootDir>/src/test-setup.ts'],
  globals: {},
  coverageDirectory: '../../../../coverage/libs/ui/shared/domain',
  reporters: [
    'default',
    [
      'jest-junit',
      {
        outputDirectory: 'coverage/libs/ui/shared/domain',
        outputName: 'jest-junit.xml',
      },
    ],
  ],
  transform: {
    '^.+\\.(ts|mjs|js|html)$': [
      'jest-preset-angular',
      {
        tsconfig: '<rootDir>/tsconfig.spec.json',
        stringifyContentPathRegex: '\\.(html|svg)$',
      },
    ],
  },
  transformIgnorePatterns: ['node_modules/(?!.*\\.mjs$|@ngneat/effects|@ngneat/effects-ng)'],
  snapshotSerializers: [
    'jest-preset-angular/build/serializers/no-ng-attributes',
    'jest-preset-angular/build/serializers/ng-snapshot',
    'jest-preset-angular/build/serializers/html-comment',
  ],
};
