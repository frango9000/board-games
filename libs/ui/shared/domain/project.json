{
  "name": "ui-shared-domain",
  "$schema": "../../../../node_modules/nx/schemas/project-schema.json",
  "projectType": "library",
  "sourceRoot": "libs/ui/shared/domain/src",
  "prefix": "app",
  "targets": {
    "clean": {
      "executor": "nx:run-commands",
      "options": {
        "commands": ["rimraf ./coverage/libs/ui/shared/domain"]
      }
    },
    "test": {
      "executor": "@nx/jest:jest",
      "outputs": ["{workspaceRoot}/coverage/libs/ui/shared/domain"],
      "options": {
        "jestConfig": "libs/ui/shared/domain/jest.config.ts"
      }
    },
    "lint": {
      "executor": "@nx/eslint:lint",
      "options": {
        "lintFilePatterns": ["libs/ui/shared/domain/src/**/*.ts", "libs/ui/shared/domain/src/**/*.html"]
      }
    }
  },
  "tags": []
}
