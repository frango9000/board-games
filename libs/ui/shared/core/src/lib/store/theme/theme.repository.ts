import { Injectable } from '@angular/core';
import { createStore, select, setProps, withProps } from '@ngneat/elf';
import { localStorageStrategy, persistState } from '@ngneat/elf-persist-state';

export interface ColorProps {
  primary?: string;
  accent?: string;
  warn?: string;
}

export interface ThemeProps {
  darkMode?: boolean;
  colors?: ColorProps;
}

export const themeStore = createStore({ name: 'theme' }, withProps<ThemeProps>({}));

persistState(themeStore, { key: 'theme', storage: localStorageStrategy });

@Injectable({ providedIn: 'root' })
export class ThemeRepository {
  darkMode$ = themeStore.pipe(select((state) => state.darkMode));
  colors$ = themeStore.pipe(select((state) => state.colors));

  public updateDarkMode(darkMode: boolean): void {
    themeStore.update(setProps({ darkMode }));
  }

  public updateColors(colors?: ColorProps): void {
    themeStore.update(setProps({ colors }));
  }
}
