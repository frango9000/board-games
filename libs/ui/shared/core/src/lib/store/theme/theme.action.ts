import { IThemeModel } from '@app/ui/shared/domain';
import { actionsFactory, props } from '@ngneat/effects';

const sessionActions = actionsFactory('Theme');

export const updateDarkMode = sessionActions.create('Update Dark Mode', props<{ darkMode?: boolean }>());
export const updateTheme = sessionActions.create('Update Theme', props<IThemeModel>());
