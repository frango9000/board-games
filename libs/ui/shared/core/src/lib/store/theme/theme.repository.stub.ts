import { Injectable } from '@angular/core';
import { BehaviorSubject } from 'rxjs';
import { ThemeRepository } from './theme.repository';

@Injectable({ providedIn: 'root' })
export class StubThemeRepository implements Partial<ThemeRepository> {
  darkMode$ = new BehaviorSubject<boolean>(false);

  public updateDarkMode(darkMode: boolean): void {
    this.darkMode$.next(darkMode);
  }
}

export const stubThemeRepositoryProvider = {
  provide: ThemeRepository,
  useClass: StubThemeRepository,
};
