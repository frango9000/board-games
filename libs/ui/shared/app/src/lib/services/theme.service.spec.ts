import { HttpTestingController } from '@angular/common/http/testing';
import { TestBed } from '@angular/core/testing';
import { ThemeModel } from '@app/ui/shared/domain';
import { HalFormClientTestingModule, HalFormService } from '@hal-form-client';
import { noop } from 'rxjs';
import { ThemeService } from './theme.service';

describe('ThemeService', () => {
  let service: ThemeService;
  let http: HttpTestingController;

  beforeEach(() => {
    TestBed.configureTestingModule({
      imports: [HalFormClientTestingModule],
    });
    service = TestBed.inject(ThemeService);
    http = TestBed.inject(HttpTestingController);
    TestBed.inject(HalFormService).setResource({
      _links: {
        self: { href: '/api' },
        theme: { href: '/api/theme' },
      },
    });
  });

  it('should be created', () => {
    expect(service).toBeTruthy();
  });

  it('should get theme', () => {
    const mockTheme = {
      primaryColor: '#00FFFF',
    };
    service.getTheme().subscribe((theme) => {
      expect(theme).toEqual(mockTheme);
    });
    const testRequest = http.expectOne('/api/theme');
    expect(testRequest.request.method).toEqual('GET');
    testRequest.flush(mockTheme);
  });

  it('should update theme', () => {
    const mockTheme = {
      primaryColor: '#00FFFF',
    };
    const themeResource = new ThemeModel({
      primaryColor: '#00FFFF',
      _templates: {
        update: { method: 'PATCH', target: '/api/theme' },
      },
    });
    service.updateTheme(themeResource, mockTheme).subscribe((theme) => {
      expect(theme).toEqual(mockTheme);
    });
    const testRequest = http.expectOne('/api/theme');
    expect(testRequest.request.method).toEqual('PATCH');
    testRequest.flush(mockTheme);
  });

  it('should initialize app colors with theme', () => {
    const updateAppColorsSpy = jest.spyOn(service, 'updateAppColors').mockImplementationOnce(noop);
    const mockTheme = {
      primaryColor: '#00FFFF',
    };
    service.initializeTheme().subscribe((theme) => {
      expect(theme).toEqual(mockTheme);
      expect(updateAppColorsSpy).toHaveBeenCalledTimes(1);
      expect(updateAppColorsSpy).toHaveBeenCalledWith(mockTheme);
    });
    const testRequest = http.expectOne('/api/theme');
    expect(testRequest.request.method).toEqual('GET');
    testRequest.flush(mockTheme);
  });

  it('should update app primary colors', () => {
    const documentSetPropertySpy = jest
      .spyOn(document.documentElement.style, 'setProperty')
      .mockImplementationOnce(noop);
    const mockTheme = {
      primaryColor: '#2196f3',
      accentColor: '#8bc34a',
      warnColor: '#f44336',
    };
    service.updateAppColors(mockTheme);
    expect(documentSetPropertySpy).toHaveBeenCalledTimes(84);
    expect(documentSetPropertySpy.mock.calls).toMatchSnapshot();
  });
});
