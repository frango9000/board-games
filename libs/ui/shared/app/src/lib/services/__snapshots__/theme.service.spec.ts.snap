// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ThemeService should update app primary colors 1`] = `
Array [
  Array [
    "--primary-50",
    "#e4f2fe",
  ],
  Array [
    "--primary-100",
    "#bce0fb",
  ],
  Array [
    "--primary-200",
    "#90cbf9",
  ],
  Array [
    "--primary-300",
    "#64b6f7",
  ],
  Array [
    "--primary-400",
    "#42a6f5",
  ],
  Array [
    "--primary-500",
    "#2196f3",
  ],
  Array [
    "--primary-600",
    "#1d8ef2",
  ],
  Array [
    "--primary-700",
    "#1883f0",
  ],
  Array [
    "--primary-800",
    "#1479ee",
  ],
  Array [
    "--primary-900",
    "#0b68eb",
  ],
  Array [
    "--primary-A100",
    "#639cff",
  ],
  Array [
    "--primary-A200",
    "#307cff",
  ],
  Array [
    "--primary-A400",
    "#005dfc",
  ],
  Array [
    "--primary-A700",
    "#0053e2",
  ],
  Array [
    "--primary-contrast-50",
    "rgba(0, 0, 0, 0.87)",
  ],
  Array [
    "--primary-contrast-100",
    "rgba(0, 0, 0, 0.87)",
  ],
  Array [
    "--primary-contrast-200",
    "rgba(0, 0, 0, 0.87)",
  ],
  Array [
    "--primary-contrast-300",
    "rgba(0, 0, 0, 0.87)",
  ],
  Array [
    "--primary-contrast-400",
    "rgba(0, 0, 0, 0.87)",
  ],
  Array [
    "--primary-contrast-500",
    "white",
  ],
  Array [
    "--primary-contrast-600",
    "white",
  ],
  Array [
    "--primary-contrast-700",
    "white",
  ],
  Array [
    "--primary-contrast-800",
    "white",
  ],
  Array [
    "--primary-contrast-900",
    "white",
  ],
  Array [
    "--primary-contrast-A100",
    "rgba(0, 0, 0, 0.87)",
  ],
  Array [
    "--primary-contrast-A200",
    "white",
  ],
  Array [
    "--primary-contrast-A400",
    "white",
  ],
  Array [
    "--primary-contrast-A700",
    "white",
  ],
  Array [
    "--accent-50",
    "#f1f8e9",
  ],
  Array [
    "--accent-100",
    "#dcedc9",
  ],
  Array [
    "--accent-200",
    "#c5e1a5",
  ],
  Array [
    "--accent-300",
    "#aed580",
  ],
  Array [
    "--accent-400",
    "#9ccc65",
  ],
  Array [
    "--accent-500",
    "#8bc34a",
  ],
  Array [
    "--accent-600",
    "#83bd43",
  ],
  Array [
    "--accent-700",
    "#78b53a",
  ],
  Array [
    "--accent-800",
    "#6eae32",
  ],
  Array [
    "--accent-900",
    "#5ca122",
  ],
  Array [
    "--accent-A100",
    "#86ff2a",
  ],
  Array [
    "--accent-A200",
    "#6af600",
  ],
  Array [
    "--accent-A400",
    "#54c300",
  ],
  Array [
    "--accent-A700",
    "#49aa00",
  ],
  Array [
    "--accent-contrast-50",
    "rgba(0, 0, 0, 0.87)",
  ],
  Array [
    "--accent-contrast-100",
    "rgba(0, 0, 0, 0.87)",
  ],
  Array [
    "--accent-contrast-200",
    "rgba(0, 0, 0, 0.87)",
  ],
  Array [
    "--accent-contrast-300",
    "rgba(0, 0, 0, 0.87)",
  ],
  Array [
    "--accent-contrast-400",
    "rgba(0, 0, 0, 0.87)",
  ],
  Array [
    "--accent-contrast-500",
    "rgba(0, 0, 0, 0.87)",
  ],
  Array [
    "--accent-contrast-600",
    "rgba(0, 0, 0, 0.87)",
  ],
  Array [
    "--accent-contrast-700",
    "rgba(0, 0, 0, 0.87)",
  ],
  Array [
    "--accent-contrast-800",
    "rgba(0, 0, 0, 0.87)",
  ],
  Array [
    "--accent-contrast-900",
    "white",
  ],
  Array [
    "--accent-contrast-A100",
    "rgba(0, 0, 0, 0.87)",
  ],
  Array [
    "--accent-contrast-A200",
    "rgba(0, 0, 0, 0.87)",
  ],
  Array [
    "--accent-contrast-A400",
    "rgba(0, 0, 0, 0.87)",
  ],
  Array [
    "--accent-contrast-A700",
    "white",
  ],
  Array [
    "--warn-50",
    "#fee8e7",
  ],
  Array [
    "--warn-100",
    "#fcc7c3",
  ],
  Array [
    "--warn-200",
    "#faa19b",
  ],
  Array [
    "--warn-300",
    "#f77b72",
  ],
  Array [
    "--warn-400",
    "#f65f54",
  ],
  Array [
    "--warn-500",
    "#f44336",
  ],
  Array [
    "--warn-600",
    "#f33d30",
  ],
  Array [
    "--warn-700",
    "#f13429",
  ],
  Array [
    "--warn-800",
    "#ef2c22",
  ],
  Array [
    "--warn-900",
    "#ec1e16",
  ],
  Array [
    "--warn-A100",
    "#ff6e69",
  ],
  Array [
    "--warn-A200",
    "#ff3d36",
  ],
  Array [
    "--warn-A400",
    "#ff0b03",
  ],
  Array [
    "--warn-A700",
    "#e90700",
  ],
  Array [
    "--warn-contrast-50",
    "rgba(0, 0, 0, 0.87)",
  ],
  Array [
    "--warn-contrast-100",
    "rgba(0, 0, 0, 0.87)",
  ],
  Array [
    "--warn-contrast-200",
    "rgba(0, 0, 0, 0.87)",
  ],
  Array [
    "--warn-contrast-300",
    "rgba(0, 0, 0, 0.87)",
  ],
  Array [
    "--warn-contrast-400",
    "rgba(0, 0, 0, 0.87)",
  ],
  Array [
    "--warn-contrast-500",
    "white",
  ],
  Array [
    "--warn-contrast-600",
    "white",
  ],
  Array [
    "--warn-contrast-700",
    "white",
  ],
  Array [
    "--warn-contrast-800",
    "white",
  ],
  Array [
    "--warn-contrast-900",
    "white",
  ],
  Array [
    "--warn-contrast-A100",
    "rgba(0, 0, 0, 0.87)",
  ],
  Array [
    "--warn-contrast-A200",
    "white",
  ],
  Array [
    "--warn-contrast-A400",
    "white",
  ],
  Array [
    "--warn-contrast-A700",
    "white",
  ],
]
`;
